src/main/java/com/rsr5/gocd/riemann_notifier/GoApiAccessor.java

```diff
@@ -7,13 +7,42 @@
 import java.io.InputStreamReader;
 import java.net.HttpURLConnection;
 import java.net.URL;
+import java.util.Base64;
 
 public class GoApiAccessor {
+    private PluginConfig pluginConfig = null;
+
+    public GoApiAccessor(PluginConfig pluginConfig) {
+        this.pluginConfig = pluginConfig;
+    }
+
+    public GoApiAccessor() {
+        this(new PluginConfig());
+    }
+
     public JsonElement get(String path) throws IOException {
-        URL url = new URL("http", "localhost", 8153, path);
-        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        HttpURLConnection con = this.getConnection(path);
         con.setRequestProperty("Accept", "application/vnd.go.cd+json");
         con.connect();
         return JsonParser.parseReader(new InputStreamReader(con.getInputStream()));
     }
+
+    public HttpURLConnection getConnection(String path) throws IOException {
+        URL url = new URL("http", "localhost", 8153, path);
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        if (this.authRequired()) {
+            con.setRequestProperty("Authorization", this.getAuthHeader());
+        }
+        return con;
+    }
+
+    private Boolean authRequired() {
```

Comment:
Looks like you don't need a big `B` Boolean here:
```suggestion
    private boolean authRequired() {
```